### PR TITLE
Add support for single_user_mode

### PR DESCRIPTION
**Pull request by NilashishC**
Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

##### SUMMARY
- Targets https://github.com/ansible-collections/ansible.netcommon/issues/73
- Add nxos cliconf specific bits for caching feature

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
plugins/cliconf/nxos.py

### CODE REVIEW

**Comment by pabelanger** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by GomathiselviS** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
Seems like the ansible/ansible depends-on PR is not being used?

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by pabelanger** [comment]
recheck

File: plugins/cliconf/nxos.py [hunk 1]
```diff
@@ -213,7 +214,7 @@ def get_config(self, source="running", format="text", flags=None):
             cmd += " ".join(to_list(flags))
         cmd = cmd.strip()
 
-        return self.send_command(cmd)
+        return self.send_command(cmd, use_cache=True)
```
Comment (pabelanger) [inline plugins/cliconf/nxos.py:217]: So, this is going to be an API change with ansible-base, which means we'll need to update the minimal version of ansible we are going to support.

Additionally, will this feature be backported into 2.10 and 2.9? If now, we likely need to make this feature flag condition in our collection.

**Comment by pabelanger** [comment]
> Seems like the ansible/ansible depends-on PR is not being used?

I've fixed it, there was a bug in our ansible-zuul-jobs repo for ansible-test

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

File: plugins/cliconf/nxos.py (hunk 1 above)
Comment (ganeshrn) [reply plugins/cliconf/nxos.py:217]: In order to reduce dependency on ansible-base we can copy the send_command API in network_cli connection plugin which is what the PR https://github.com/ansible-collections/ansible.netcommon/pull/120/files is doing so, in that case, we won't require backport or minimal ansible version dependency to be updated.

Comment (pabelanger) [reply plugins/cliconf/nxos.py:217]: ++ I think that is the right approach, I guess we'll have to remove the bits in ansible/ansible right?

Comment (ganeshrn) [reply plugins/cliconf/nxos.py:217]: Yes, it can be removed, but need to be careful doing so in order to ensure backward compatibility is maintained.

**Comment by NilashishC** [comment]
recheck

File: plugins/cliconf/nxos.py [hunk 2]
```diff
@@ -28,6 +28,15 @@
 - This nxos plugin provides low level abstraction apis for sending and receiving CLI
   commands from Cisco NX-OS network devices.
 version_added: 1.0.0
+options:
+  config_commands:
+    description:
+    - Specifies a list of commands that can make configuration changes
```
Comment (ganeshrn) [inline plugins/cliconf/nxos.py:34]: The description can be more verbose as to why, where and how it is used.

File: plugins/cliconf/nxos.py [hunk 3]
```diff
@@ -28,6 +28,15 @@
 - This nxos plugin provides low level abstraction apis for sending and receiving CLI
   commands from Cisco NX-OS network devices.
 version_added: 1.0.0
+options:
+  config_commands:
```
Comment (ganeshrn) [inline plugins/cliconf/nxos.py:32]: also, add version_added string

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
Scheduled to be merged and released alongside ansible.netcommon v2.0.0.

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by NilashishC** [comment]
recheck

**Comment by ashwini-mhatre** [comment]
LGTM

**Pull request merged**